Assets/_Project/Scripts/Map/DeviceLocalizationProvider.cs

```diff
@@ -0,0 +1,351 @@
+namespace PolSl.UrbanHealthPath.Map
+{
+
+
+	using System.Collections;
+	using UnityEngine;
+	using Mapbox.Utils;
+	using Mapbox.CheapRulerCs;
+	using System;
+	using System.Linq;
+	using Mapbox.Unity.Location;
+
+
+	/// <summary>
+	/// The DeviceLocationProvider is responsible for providing real world location and heading data,
+	/// served directly from native hardware and OS. 
+	/// This relies on Unity's <see href="https://docs.unity3d.com/ScriptReference/LocationService.html">LocationService</see> for location
+	/// and <see href="https://docs.unity3d.com/ScriptReference/Compass.html">Compass</see> for heading.
+	/// </summary>
+	public class DeviceLocalizationProvider : AbstractLocationProvider, ILocalizationProvider
```

Comment:
This class definitely does way too many things. Write a custom one and include only details that we really need.